ModelBase/ModelBase.pro

```diff
@@ -50,7 +50,9 @@ HEADERS += src/persistence/NodeIdMap.h \
     test/PersistentStoreMock.h \
     src/ModelBasePlugin.h \
     src/model/TreeManager.h \
-    src/model/AllTreeManagers.h
+    src/model/AllTreeManagers.h \
+    src/nodes/NodeReference.h \
+    src/nodes/EmptyNode.h
```

Comment:
So, I have quite a few things to say here:
- We don't need two classes I think. Only one should be enough. It may hold a reference to a node or may not hold a reference, in which case it can be used as a spacer.
- Whatever the case, this class/classes are specific to your functionality. Therefore they should be in the VisualizationBase project and not in ModelBase. Please move all node related code in VisualizationBase/src/nodes which already exists.
- In the end I expect that there are *no* changes at all in ModelBase.